### PR TITLE
Replace WCA ID input with user search

### PR DESCRIPTION
**Pull request by danieljames-dj**
This is basically a rough code to do the basic working implementation. Can someone please suggest if there is a better way to handle single input in OmniSearch?

### CODE REVIEW

**Review by gregorbg — CHANGES_REQUESTED** [review]
Please wait for https://github.com/thewca/worldcubeassociation.org/pull/7459, which will simplify working with the Semantic search bars from `OmnisearchDropdown`.

Your suggested implementation unfortunately has a few flaws (most notably, you're relying on inline CSS for components that should be shared across a whole wide range of different layouts, which is very impractical!) so I'll just block this until the aforementioned PR is merged.

**Comment by danieljames-dj** [comment]
Okay, will wait for that PR to be merged.

**Comment by danieljames-dj** [comment]
Marking this PR as blocked, awaiting for https://github.com/thewca/worldcubeassociation.org/pull/7459 to merge.

**Comment by danieljames-dj** [comment]
@gregorbg Since Board has requested for this feature in probation page (mainly because it blocks the case where there is delegate without WCA ID), I've brought the corresponding code from PR #7459 to this PR. Could you please review again if this is fine?

File: WcaOnRails/app/webpacker/components/DelegateProbations/index.jsx
Comment (gregorbg) [inline WcaOnRails/app/webpacker/components/DelegateProbations/index.jsx:88]: The `setUserId` action should be part of the success action in the `save` invocation (the part that currently forwards to `sync`)

**Review by gregorbg — CHANGES_REQUESTED** [review]
Tiny change about the success hook after saving, otherwise fine!

File: WcaOnRails/app/webpacker/components/DelegateProbations/index.jsx
Comment (danieljames-dj) [reply WcaOnRails/app/webpacker/components/DelegateProbations/index.jsx:88]: Okay, done.

**Review by gregorbg — APPROVED** [review]

**Pull request merged**
